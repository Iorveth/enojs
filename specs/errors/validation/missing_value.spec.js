const eno = require('../../../eno.js');

describe('validation.missingValue', () => {

  let error;

  describe('missingFieldsetEntryValue', () => {
    beforeAll(() => {
      const document = eno.parse(`
        fieldset:
        required =
      `);

      try {
        document.fieldset('fieldset').entry('required', { required: true });
      } catch(err) {
        error = err;
      }
    })

    it('provides a correct message', () => {
      expect(error.message).toMatchSnapshot();
    });

    it('provides correct selection metadata', () => {
      expect(error.selection).toMatchSnapshot();
    });
  });

  describe('missingFieldValue', () => {
    describe('without empty line continuations', () => {
      beforeAll(() => {
        const document = eno.parse(`
          required:
        `);

        try {
          document.field('required', { required: true });
        } catch(err) {
          error = err;
        }
      })

      it('provides a correct message', () => {
        expect(error.message).toMatchSnapshot();
      });

      it('provides correct selection metadata', () => {
        expect(error.selection).toMatchSnapshot();
      });
    });

    describe('with empty line continuations', () => {
      beforeAll(() => {
        const document = eno.parse(`
          required:
          |

          |
        `);

        try {
          document.field('required', { required: true });
        } catch(err) {
          error = err;
        }
      })

      it('provides a correct message', () => {
        expect(error.message).toMatchSnapshot();
      });

      it('provides correct selection metadata', () => {
        expect(error.selection).toMatchSnapshot();
      });
    });
  });

  describe('missingListItemValue', () => {
    beforeAll(() => {
      const document = eno.parse(`
        values required:
        - value
        -
      `);

      try {
        document.list('values required'); // { enforceValues: true } is default
      } catch(err) {
        error = err;
      }
    })

    it('provides a correct message', () => {
      expect(error.message).toMatchSnapshot();
    });

    it('provides correct selection metadata', () => {
      expect(error.selection).toMatchSnapshot();
    });
  });
});
