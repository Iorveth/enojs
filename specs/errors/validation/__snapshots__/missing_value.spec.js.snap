// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation.missingValue missingDictionaryEntryValue provides a correct message 1`] = `
"The dictionary entry 'required' must contain a value.

   Line | Content
      1 | 
      2 |         dictionary:
 >    3 |         required =
      4 |       
"
`;

exports[`validation.missingValue missingDictionaryEntryValue provides correct selection metadata 1`] = `
Array [
  Array [
    2,
    18,
  ],
  Array [
    2,
    18,
  ],
]
`;

exports[`validation.missingValue missingFieldValue with empty appendices provides a correct message 1`] = `
"The field 'required' must contain a value.

   Line | Content
      1 | 
 >    2 |           required:
 *    3 |           |
 *    4 | 
 *    5 |           |
      6 |         
"
`;

exports[`validation.missingValue missingFieldValue with empty appendices provides correct selection metadata 1`] = `
Array [
  Array [
    1,
    19,
  ],
  Array [
    4,
    11,
  ],
]
`;

exports[`validation.missingValue missingFieldValue without empty appendices provides a correct message 1`] = `
"The field 'required' must contain a value.

   Line | Content
      1 | 
 >    2 |           required:
      3 |         
"
`;

exports[`validation.missingValue missingFieldValue without empty appendices provides correct selection metadata 1`] = `
Array [
  Array [
    1,
    19,
  ],
  Array [
    1,
    19,
  ],
]
`;

exports[`validation.missingValue missingListItemValue provides a correct message 1`] = `
"The list 'values required' may not contain empty items.

   Line | Content
   ...
      2 |         values required:
      3 |         - value
 >    4 |         -
      5 |       
"
`;

exports[`validation.missingValue missingListItemValue provides correct selection metadata 1`] = `
Array [
  Array [
    3,
    9,
  ],
  Array [
    3,
    9,
  ],
]
`;
